src/api/app/controllers/comments_controller.rb

```diff
@@ -31,7 +31,9 @@ def find_obj
         @header = { project: @obj.name }
       end
     elsif params[:request_number]
-      @obj = BsRequest.find_by_number!(params[:request_number])
+      @obj = BsRequest.find_by(number: params[:request_number])
+      raise ActiveRecord::RecordNotFound, "Couldn't find Request with number '#{params[:request_number]}'" if @obj.nil?
```

Comment:
I'd even drop (here and in the other places) the `with` word, as `number` is not a property or a detail but it is the identifier of the `Request`

`Couldn't find Request number 1234` looks fine by me. What do you think?

Reply:
I didn't want to deviate much from the by default message `"Couldn't find #{name} with '#{primary_key}'=#{id}"`. See https://github.com/rails/rails/blob/869459ed0c9eb368ad88225d056f6d42adffc9d4/activerecord/lib/active_record/core.rb#L252

I would prefer to keep the `with`.

Reply:
Oh, I see. Consistency for the win!!
Fine by me then :+1: